recipe_scrapers/_schemaorg.py

```diff
@@ -46,27 +50,31 @@ def __init__(self, page_data):
                                 self.data = graph_item
                                 return
 
-    def language(self):
+    def language(self) -> Optional[str]:
         return self.data.get("inLanguage") or self.data.get("language")
 
-    def title(self):
+    def title(self) -> Optional[str]:
         return normalize_string(self.data.get("name"))
 
-    def author(self):
+    def author(self) -> Optional[str]:
         author = self.data.get("author")
         if author and isinstance(author, dict):
             author = author.get("name")
         return author
 
-    def total_time(self):
+    def total_time(self) -> Optional[int]:
         total_time = get_minutes(self.data.get("totalTime"))
-        if not total_time:
-            prep_time = get_minutes(self.data.get("prepTime")) or 0
-            cook_time = get_minutes(self.data.get("cookTime")) or 0
-            total_time = prep_time + cook_time
+        if total_time is None:
+            times = [get_minutes(self.data.get(p)) for p in ["prepTime", "cookTime"]]
+
+            total_time = (
+                None
+                if all([b is None for b in times])
+                else sum([t for t in times if t is not None])
+            )
```

Comment:
Idea: could potentially be simplified to:

```py
sum([t for t in times if t]) or None
```

Reply:
I went to bed thinking about that code and I think it caused my nightmares.

I'm going to redo it later today, but we have to be careful about values being `falsey`. I want to return zero if the website has zeros in it, but if the website has `null` or is empty then we can return `None`. I'll add tests for this.

```python
>>> times = [None, None] # should return None
>>> sum([t for t in times if t]) or None
>>> times = [0, None] # should return 0
>>> sum([t for t in times if t]) or None
>>> times = [0, 0] # should return 0
>>> sum([t for t in times if t]) or None
>>> sum([t for t in times if t is not None]) or None
>>> sum([t for t in times if t is not None])
0 # desired result since website "had" zeroes.
>>> sum([])
0 # bad. I need to check if a filtered times is empty
```